### ISSUE TITLE
glvnd enabled mesa - undefined symbol with intel hw

### ISSUE THREAD

**Issue by kwizart** [NVIDIA/libglvnd#104]
I'm testing a glvnd enabled version of mesa 12.0.3 on fedora 25 intel device.
But this lead to have few undefined symbols

```
fbcon: inteldrmfb (fb0) is primary device
Console: switching to colour frame buffer device 170x48
i915 0000:00:02.0: fb0: inteldrmfb frame buffer device
[...]
(==) Depth 24 pixmap format is 32 bpp
(II) intel(0): SNA initialized with Alviso (gen3) backend
(==) intel(0): Backing store enabled
(==) intel(0): Silken mouse enabled
(II) intel(0): HW Cursor enabled
(II) intel(0): RandR 1.2 enabled, ignore the following RandR disabled message.
(==) intel(0): DPMS enabled
(==) intel(0): Display hotplug detection enabled
(II) intel(0): [XvMC] i915_xvmc driver initialized.
(II) intel(0): [DRI2] Setup complete
(II) intel(0): [DRI2]   DRI driver: i915
(II) intel(0): [DRI2]   VDPAU driver: va_gl
(II) intel(0): direct rendering: DRI2 enabled
(II) intel(0): hardware support for Present enabled
(--) RandR disabled
(II) SELinux: Disabled by boolean
(EE) AIGLX error: dlopen of /usr/lib64/dri/i915_dri.so failed (/usr/lib64/dri/i915_dri.so: undefined symbol: _glapi_get_dispatch_table_size)
(EE) AIGLX: reverting to software rendering
(EE) AIGLX error: dlopen of /usr/lib64/dri/swrast_dri.so failed (/usr/lib64/dri/swrast_dri.so: undefined symbol: _glapi_check_multithread)
(EE) GLX: could not load software renderer
(II) GLX: no usable GL providers found for screen 0
(II) intel(0): switch to mode 1366x768@60.0 on LVDS1 using pipe 1, position (0, 0), rotation normal, reflection none
(II) intel(0): Setting screen physical size to 361 x 203

```

**Comment by kbrenneman** [comment]
Those are symbols in Mesa, not libglvnd.

_glapi_get_dispatch_table_size is an internal symbol in libGLdispatch.so (inherited from Mesa's glapi code), but it's not exported.

_glapi_check_multithread is only in Mesa, though, as the equivalent code in libglvnd was rewritten quite a while ago.

**Comment by kwizart** [comment]
Thx for your reply.

It's probably a mesa issue, but It's only triggered by using a glvnd enabled mesa.
In both cases, there are undefined non-weak-symbols as shown:

> $ ldd  -r /usr/lib64/dri/i915_dri.so
> 	linux-vdso.so.1 (0x00007ffef459c000)
> 	libdrm_intel.so.1 => /lib64/libdrm_intel.so.1 (0x00007f37792de000)
> 	libdrm_nouveau.so.2 => /lib64/libdrm_nouveau.so.2 (0x00007f37790d5000)
> 	libdrm_radeon.so.1 => /lib64/libdrm_radeon.so.1 (0x00007f3778ec9000)
> 	libselinux.so.1 => /lib64/libselinux.so.1 (0x00007f3778ca2000)
> 	libdrm.so.2 => /lib64/libdrm.so.2 (0x00007f3778a93000)
> 	libexpat.so.1 => /lib64/libexpat.so.1 (0x00007f3778866000)
> 	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f377864a000)
> 	libdl.so.2 => /lib64/libdl.so.2 (0x00007f3778446000)
> 	libm.so.6 => /lib64/libm.so.6 (0x00007f377813c000)
> 	libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f3777f25000)
> 	libc.so.6 => /lib64/libc.so.6 (0x00007f3777b63000)
> 	/lib64/ld-linux-x86-64.so.2 (0x00005574e1d2c000)
> 	libpciaccess.so.0 => /lib64/libpciaccess.so.0 (0x00007f3777958000)
> 	libpcre.so.1 => /lib64/libpcre.so.1 (0x00007f37776e5000)
> undefined symbol: _glapi_get_dispatch_table_size	(/usr/lib64/dri/i915_dri.so)
> undefined symbol: _glapi_get_context	(/usr/lib64/dri/i915_dri.so)
> undefined symbol: _glapi_add_dispatch	(/usr/lib64/dri/i915_dri.so)
> undefined symbol: _glapi_check_multithread	(/usr/lib64/dri/i915_dri.so)
> undefined symbol: _glapi_tls_Context	(/usr/lib64/dri/i915_dri.so)
> undefined symbol: _glapi_set_context	(/usr/lib64/dri/i915_dri.so)
> undefined symbol: _glapi_set_dispatch	(/usr/lib64/dri/i915_dri.so)
> undefined symbol: _glapi_tls_Dispatch	(/usr/lib64/dri/i915_dri.so)

In mesa glvnd version, libGLX_mesa.so.0 is linked to libglapi whereas in non-glvnd version, it's libGL.so.1.

I'm trying to push as a fix in mesa in this report
https://bugs.freedesktop.org/show_bug.cgi?id=98428

Thx

**Comment by kwizart** [comment]
This issue has been fixed in upstream mesa 17.1x and 17.0.6
